x/distribution/types/fee_pool.go

```diff
@@ -0,0 +1,50 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// total accumulation tracker
+type TotalAccum struct {
+	UpdateHeight int64   `json:"update_height"`
+	Accum        sdk.Dec `json:"accum"`
+}
+
+func NewTotalAccum(height int64) TotalAccum {
+	return TotalAccum{
+		UpdateHeight: height,
+		Accum:        sdk.ZeroDec(),
+	}
+}
+
+// update total validator accumulation factor
+func (ta TotalAccum) Update(height int64, accumCreatedPerBlock sdk.Dec) TotalAccum {
```

Comment:
`UpdateForNewHeight` maybe - also, looks like height must be greater than the current height, let's mention that

Reply:
added also - added a sanity checks